pyramid/scripts/pcreate.py

```diff
@@ -21,73 +22,76 @@ def main(argv=sys.argv, quiet=False):
+    parser.add_argument('-l', '--list',
+                        dest='list',
+                        action='store_true',
+                        help="List all available scaffold names")
+    parser.add_argument('--list-templates',
+                        dest='list',
+                        action='store_true',
+                        help=("A backwards compatibility alias for -l/--list. "
+                              "List all available scaffold names."))
+    parser.add_argument('--package-name',
+                        dest='package_name',
+                        action='store',
+                        help='Package name to use. The name provided is '
+                             'assumed to be a valid Python package name, and '
+                             'will not be validated. By default the package '
+                             'name is derived from the value of '
+                             'output_directory.')
+    parser.add_argument('--simulate',
+                        dest='simulate',
+                        action='store_true',
+                        help='Simulate but do no work')
+    parser.add_argument('--overwrite',
+                        dest='overwrite',
+                        action='store_true',
+                        help='Always overwrite')
+    parser.add_argument('--interactive',
+                        dest='interactive',
+                        action='store_true',
+                        help='When a file would be overwritten, interrogate '
+                             '(this is the default, but you may specify it to '
+                             'override --overwrite)')
+    parser.add_argument('--ignore-conflicting-name',
+                        dest='force_bad_name',
+                        action='store_true',
+                        default=False,
+                        help='Do create a project even if the chosen name '
+                             'is the name of an already existing / importable '
+                             'package.')
+    parser.add_argument('output_directory',
+                        help='The directory where the project will be '
+                             'created.')
 
     pyramid_dist = pkg_resources.get_distribution("pyramid")
 
     def __init__(self, argv, quiet=False):
         self.quiet = quiet
-        self.options, self.args = self.parser.parse_args(argv[1:])
-        if not self.options.interactive and not self.options.overwrite:
-            self.options.interactive = True
+        self.args = self.parser.parse_args(argv[1:])
+        if not self.args.interactive and not self.args.overwrite:
+            self.args.interactive = True
         self.scaffolds = self.all_scaffolds()
 
     def run(self):
-        if self.options.list:
+        if self.args.list:
             return self.show_scaffolds()
-        if not self.options.scaffold_name and not self.args:
+        if not self.args.scaffold_name and not self.args:
```

Comment:
`self.args` is not what it used to be, you really want to test for `self.args.output_directory`. Also, this should never be falsy because `argparse` requires the user to provide input unless you make it optional.

In `optparse`:

- `self.options`: Contained named options/flags
- `self.args`: Contained whatever arguments were left over (positional arguments)

in `argparse` optional arguments are now rolled into `self.args` as well, and providing more arguments than expected will cause `argparse` to call sys.exit while providing the help text.